server/venueless/api/views.py

```diff
@@ -115,6 +119,90 @@ def get(self, request, **kwargs):
             )
 
 
+class CreateWorldView(APIView):
+    authentication_classes = []  # disables authentication
+    permission_classes = []
+
+    @staticmethod
+    def post(request, *args, **kwargs) -> JsonResponse:
+        payload = CreateWorldView.get_payload_from_token(request)
+
+        # check if user has permission to create world
+        if payload.get("has_permission"):
+            secret = get_random_string(length=64)
+            config = {
+                "JWT_secrets": [
+                    {
+                        "issuer": "any",
+                        "audience": "venueless",
+                        "secret": secret,
+                    }
+                ]
+            }
+
+            title_default = {
+                value for key, value in request.data.get("title").items() if value
+            }.pop()
+
+            # if world already exists, update it, otherwise create a new world
+            try:
+                if World.objects.filter(id=request.data.get("id")).exists():
+                    world = World.objects.get(id=request.data.get("id"))
+                    world.title = (
+                        request.data.get("title")[request.data.get("locale")]
+                        or request.data.get("title")["en"]
+                        or title_default
+                    )
+                    world.domain = "{}{}/{}".format(
+                        settings.DOMAIN_PATH, settings.BASE_PATH, request.data.get("id")
+                    )
+                    world.locale = request.data.get("locale")
+                    world.timezone = request.data.get("timezone")
+                    world.save()
+                else:
+                    world = World.objects.create(
+                        id=request.data.get("id"),
+                        title=request.data.get("title")[request.data.get("locale")]
+                        or request.data.get("title")["en"]
```

Comment:
What if `data(get('title'))` returns `None` or something not a `dict`. Validation needs to be applied to make sure the incoming data has the expected data type and shape.